exporters/metric/cortex/cortex.go

```diff
@@ -86,11 +109,228 @@ func InstallNewPipeline(config Config, options ...push.Option) (*push.Controller
+			tSeries, err := convertFromMinMaxSumCount(record, minMaxSumCount)
+			if err != nil {
+				return err
+			}
+
+			timeSeries = append(timeSeries, tSeries...)
+		} else if lastValue, ok := agg.(aggregation.LastValue); ok {
+			tSeries, err := convertFromLastValue(record, lastValue)
+			if err != nil {
+				return err
+			}
+
+			timeSeries = append(timeSeries, tSeries)
+		}
+
+		return nil
+	})
+
+	// Check if error was returned in checkpointSet.ForEach()
+	if aggError != nil {
+		return nil, aggError
+	}
+
+	return timeSeries, nil
+}
+
+// convertFromSum returns a single TimeSeries based on a Record with a Sum aggregation
+func convertFromSum(record metric.Record, sum aggregation.Sum) (*prompb.TimeSeries, error) {
+	// Get Sum value
+	value, err := sum.Sum()
+	if err != nil {
+		return nil, err
+	}
+	// Create sample from Sum value
+	sample := prompb.Sample{
+		Value:     value.CoerceToFloat64(record.Descriptor().NumberKind()),
+		Timestamp: record.EndTime().Unix(), // Convert time to Unix (int64)
+	}
+
+	// Create labels, including metric name
+	name := sanitize(record.Descriptor().Name())
+	labels := createLabelSet(record, "__name__", name)
+
+	// Create TimeSeries and return
+	tSeries := &prompb.TimeSeries{
+		Samples: []prompb.Sample{sample},
+		Labels:  labels,
+	}
+
+	return tSeries, nil
+}
+
+// convertFromLastValue returns a single TimeSeries based on a Record with a LastValue aggregation
+func convertFromLastValue(record metric.Record, lastValue aggregation.LastValue) (*prompb.TimeSeries, error) {
+	// Get value
+	value, _, err := lastValue.LastValue()
+	if err != nil {
+		return nil, err
+	}
+
+	// Create sample from Last value
+	sample := prompb.Sample{
+		Value:     value.CoerceToFloat64(record.Descriptor().NumberKind()),
+		Timestamp: record.EndTime().Unix(), // Convert time to Unix (int64)
+	}
+
+	// Create labels, including metric name
+	name := sanitize(record.Descriptor().Name())
+	labels := createLabelSet(record, "__name__", name)
+
+	// Create TimeSeries and return
+	tSeries := &prompb.TimeSeries{
+		Samples: []prompb.Sample{sample},
+		Labels:  labels,
+	}
+
+	return tSeries, nil
+}
+
+// convertFromMinMaxSumCount returns 4 TimeSeries for the min, max, sum, and count from the mmsc aggregation
+func convertFromMinMaxSumCount(record metric.Record, minMaxSumCount aggregation.MinMaxSumCount) ([]*prompb.TimeSeries, error) {
+	// Convert Min
+	min, err := minMaxSumCount.Min()
+	if err != nil {
+		return nil, err
+	}
+	minSample := prompb.Sample{
+		Value:     min.CoerceToFloat64(record.Descriptor().NumberKind()),
+		Timestamp: record.EndTime().Unix(), // Convert time to Unix (int64)
+	}
+
+	// Create labels, including metric name
+	name := sanitize(record.Descriptor().Name() + "_min")
+	labels := createLabelSet(record, "__name__", name)
+
+	// Create TimeSeries
+	minTimeSeries := &prompb.TimeSeries{
+		Samples: []prompb.Sample{minSample},
+		Labels:  labels,
+	}
+
+	// Convert Max
```

Comment:
I wonder if you'd consider refactoring this so the three blocks of nearly-identical code (for Min, Max, Count)?

Reply:
Yeah, I think creating a helper function like createTimeSeries would be helpful. Maybe pass in the value I want and the labels. I'll make sure I can use it with histogram and distribution buckets/quantiles as well since that'll be pretty repetitive too.